paramiko_jump/util.py

```diff
@@ -0,0 +1,129 @@
+"""
+Utility functions and classes designed to help simplify paramiko jump as well as add extra features.
+
+"""
+from getpass import getpass
+
+from contextlib import contextmanager
+from typing import (
+    AnyStr,
+    Tuple,
+    Iterator,
+    Sequence,
+    Optional,
+    Callable,
+)
+from paramiko_jump.client import SSHJumpClient
+
+import paramiko
+
+
+Prompt = Tuple[AnyStr, bool]
+Optional[Callable] = None
+
+
+class StatefulAuthHandler:
+    """Stateful auth handler for paramiko that will return a list of auth
+     parameters for every CLI prompt
+
+    Example
+    -------
+        >>> from paramiko_jump.util import StatefulAuthHandler
+        >>> handler = StatefulAuthHandler(['password'],['1'])
+        >>> handler()
+        ['password']
+        >>> handler()
+        ['1']
+
+    Attributes
+    ----------
+    _iterator: Iterator
+        Iterator to iterate through all the objects in an iteratable object
+    """
+
+    _iterator = None
+
+    def __init__(self, *items: Sequence):
```

Comment:
Don't annotate items as type Sequence -- if anything, it should be type Any. The reason for this is that an annotation of a positional argument is an annotation of what is being passed in, not of the positional type itself (this isn't an annotation for items, but rather what items contains).

Reply:
aaahhh ok that makes sense! thank you!